app/controllers/application_controller.rb

```diff
@@ -20,12 +20,14 @@ class ApplicationController < Sinatra::Base
     if params[:username] == "" || params[:password] == ""
       redirect '/failure'
     else
+      User.create(username: params[:username], password: params[:password])
```

Comment:
We should either use `.create!` or an `if`-block to make sure the User was successfully created.